sphinx_needs/data.py

```diff
@@ -167,33 +174,38 @@ class CoreFieldParameters(TypedDict):
     "type": {
         "description": "Type of the need.",
         "schema": {"type": "string", "default": ""},
+        "allow_df": True,
```

Comment:
How can the type and below type specific values be set by a df?

Reply:
well thats a discussion to be had; I wouldn't expect them to, but they technically can already, so removing them would be a breaking change